admin/providers/views.py

```diff
@@ -0,0 +1,53 @@
+import requests
+from website import settings
+from osf.models import AbstractProvider
+from django.contrib.auth.mixins import PermissionRequiredMixin
+from django.views.generic import View
+from django.urls import reverse_lazy
+from django.shortcuts import redirect
+
+
+class ShareSourceProvider(PermissionRequiredMixin, View):
+    view_name = 'share_source'
+    permission_required = 'osf.change_registrationprovider'
+
+    raise_exception = True
+
+    def get(self, request, *args, **kwargs):
+        provider = AbstractProvider.objects.get(id=list(self.kwargs.values())[0])
+
+        if provider.share_source or provider.access_token:
+            raise ValueError('Cannot update share_source or access_token because one or the other already exists')
+        if not settings.SHARE_ENABLED or not settings.SHARE_URL:
+            raise ValueError('SHARE_ENABLED is set to false')
+        if not provider.get_asset_url('square_color_no_transparent'):
+            raise ValueError('Unable to find "square_color_no_transparent" icon for provider')
+
+        resp = requests.post(
+            f'{settings.SHARE_URL}api/v2/sources/',
+            json={
+                'data': {
+                    'type': 'Source',
+                    'attributes': {
+                        'homePage': provider.domain if provider.domain else f'{settings.DOMAIN}/registration/{provider._id}/',
```

Comment:
`/registration/` here is wrong -- it'll vary by provider type: `/registries/` and `/preprints/`

ideally we could get the "home page" url from the provider model, but failing that could just build it in each view and add a second parameter to that share'd function
```py
def setup_share_source(provider, provider_home_page):
```